assignment2/assignment2.py

```diff
@@ -0,0 +1,418 @@
+    """
+    return employees["fields"].index(string)
+
+
+# Task 4
+def first_name(row_number: int) -> str:
+    """
+
+    Create a function called first_name.  It takes one argument, the row number.  The function should retrieve the value of first_name from a row as stored in the employees dict.
+
+    You should first call your column_index function to find out what column index you want.
+
+    Then you go to the requested row as stored in the employees dict, and get the value at that index in the row.
+
+    Return the value.
+
+    Try the test again.
+    """
+    name_index: int = column_index("first_name")
+    return employees["rows"][row_number][name_index]
+
+
+employee_id_column = column_index("employee_id")
+print(employees)
+
+
+# Task 5
+def employee_find(employee_id: int) -> list:
+    """
+
+
+
+        Create a function called employee_find.  This is passed one argument, an integer.  Just call it employee_id in your function declaration. We want it to return the rows with the matching employee_id.  There should only be one, but sometimes a CSV file has bad data.
+
+        We could do this with a loop.  But we are going to use the filter() function.  Inside the employee_find function (yes, you do declare functions inside functions sometimes), create the following employee_match function:
+
+    def employee_match(row):
+       return int(row[employee_id_column]) == employee_id
+
+    This function is referencing the employee_id value that is passed to the employee_find function.  It can access that value because the employee_match function is inside the employee_find function.  Note that we need to do type conversion here, because the CSV reader just returns strings as the values in the roows.  This inner function returns True if there is a match.  We are using the employee_id_column global value you set in Task 3.
+
+        Now, still within the employee_find function, call the filter() function. This is another one of those Python free standing functions. (It is not a method of the list class.) You call filter() as follows:
+
+    matches=list(filter(employee_match, employees["rows"]))
+
+    The filter() function needs to know how to filter, and the employee_match function provides that information.  The filter() function calls employee_match once per row, saying, Do we want this one?  When the filter function completes, we need to do type conversion to convert the result to a list.
+
+        The employee_find function then returns the matches.
+
+        Run the test and see if you got it right.
+
+
+    """
+
+    def employee_match(row):
+        return int(row[employee_id_column]) == employee_id
+
+    matches = list(filter(employee_match, employees["rows"]))
+    return matches
+
+
+# Task 6
+def employee_find_2(employee_id: int) -> list:
+    """
+    The employee_match function is a silly one-liner.  Lambdas allow us to give the logic inline.
+
```

Comment:
There is nothing wrong with "silly one liners" if it makes code more readable. Lambdas tend to make code less readable, less Pythonic. I know that isn't your sentiment, it is from the assignment. 😄  I'm just saying that one-line function is pretty nice, I wouldn't replace it with a lambda in real code.